packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionHierarchy_def.hpp

```diff
@@ -1314,6 +1367,10 @@ void createRegionHierarchy(const int maxRegPerProc,
                               regMatrices,
                               coarseCompOp);
 
+  std::cout << mapComp->getComm()->getRank() << " | MakeCoarseCompositeDirectSolver ..." << std::endl;
+
+  coarseSolver = MakeCompositeDirectSolver(coarseCompOp);
```

Comment:
I think we should go a step further and actually do the LU factorization here, calling the following methods:
```
coarseSolver->symbolicFactorization();
coarseSolver->numericFactorization();
```
You can see an example in Amesos2 [here](https://trilinos.org/docs/r12.12/packages/amesos2/doc/html/TwoPartSolve_8cpp-example.html).

Reply:
I call exactly these two methods inside `MakeCompositeDirectSolver()` as the last step before returning the solver object.

Reply:
Yes I see that now! Great then it's good with me : )